File: PythonProject/JudgeRouteCircle.py
class Solution:
    UpKey = 1
    RightKey = 2
    def judgeCircle(self, moves):
        n = 0
        for m in moves:
            if m == 'U':
                n += self.UpKey
            if m == 'D':
                n += -self.UpKey
            if m == 'R':
                n += self.RightKey
            if m == 'L':
                n += -self.RightKey
        return n == 0

s = Solution()
print(s.judgeCircle("URLDD"))

File: JavaProject/src/com/company/JudgeRouteCircle.java
package com.company;


public class JudgeRouteCircle {
    static int UpKey = 1;
    static int RightKey = 2;
    public boolean judgeCircle(String moves) {
        int n = 0;
        for (int i = 0; i < moves.length(); i ++) {
            if (moves.charAt(i) == 'U') {
                n += UpKey;
            }
            if (moves.charAt(i) == 'D') {
                n += -UpKey;
            }
            if (moves.charAt(i) == 'R') {
                n += RightKey;
            }
            if (moves.charAt(i) == 'L') {
                n += -RightKey;
            }
        }
        return n == 0;
    }
}
